src/Auto/AutoRoutineSetter.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * AutoRoutineSetter.cpp
+ *
+ *  Created on: Jan 29, 2017
+ *      Author: peter
+ */
+
+#include "AutoRoutineSetter.h"
+
+static Timer* autoTimer = new Timer();
+
+
+void AutoRoutineSetter::setAutoRoutine(AutoRoutine* new_auto_routine) {
+
+  m_auto_routine = new_auto_routine;
+
+}
+
+void AutoRoutineSetter::Start() {
+
+  if(m_auto_routine != nullptr) {
+
+    m_auto_routine->Run();
+  }
+  autoTimer->Start();
```

Comment:
Why call start if `m_auto_routine` is a `nullptr`? Should this be inside the if call?